app/models/remote_execution_provider.rb

```diff
@@ -44,6 +44,11 @@ def effective_user_method(host)
       method
     end
 
+    def cleanup_working_dirs?(host)
+      setting = host_setting(host, :remote_execution_cleanup_working_dirs)
+      ['true', 'True', 'TRUE', '1'].include?(setting)
```

Comment:
When the value gets taken from global settings it is a `TrueClass` or `FalseClass`, not `String` and this check then returns `false`. Setting the value on a host level seems to work correctly